File: c#/tema_3/ConcertPro/Service/UserService.cs
using System;
using ConcertPro.Dao.Exception;
using ConcertPro.Dao.Model;
using ConcertPro.Dao.Repository.UserRepository;
using log4net;

namespace ConcertPro.Service
{
    public class UserService
    {
        private static readonly ILog Log = LogManager.GetLogger("UserService");
        private readonly IUserRepository _userRepository;

        public UserService(IUserRepository userRepository)
        {
            Log.InfoFormat("Initializing the UserService.");
            
            _userRepository = userRepository;
        }

        public void LoginUser(string username, string password)
        {
            Log.InfoFormat("Entering user login function.");

            User user = _userRepository.FindByUsernameAndPassword(username, password);

            if (user == null)
            {
                throw new ValidationException("Username sau parola gresite!!");
            }
            
            user.LoggedIn = true;
            _userRepository.Update(user, user.Id);
            
            Log.InfoFormat("User logged in. {0}", user);
        } 
    }
}

File: java/tema_5/src/main/java/ro/mpp2024/concertpro/service/UserService.java
package ro.mpp2024.concertpro.service;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp2024.concertpro.dao.exception.ValidationException;
import ro.mpp2024.concertpro.dao.model.User;
import ro.mpp2024.concertpro.dao.repository.user_repository.IUserRepository;

public class UserService {
    private final IUserRepository userRepository;
    private static final Logger logger = LogManager.getLogger();

    public UserService(IUserRepository userRepository) {
        logger.info("Initializing User service.");
        this.userRepository = userRepository;
    }

    public void loginUser(String username, String password) {
        logger.traceEntry("Entering user log in function");

        User user = this.userRepository.findByUsernameAndPassword(username, password);

        if(user.getId() == 0) {
            throw new ValidationException("Wrong username or password!!");
        }

        user.setLoggedIn(true);
        this.userRepository.update(user, user.getId());

        logger.trace("User logged in.");
        logger.traceExit();
    }
}
